gello-world/src/components/login/login.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from "react";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import { FormErrors } from "./FormErrors.js";
+export class Login extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			username: "",
+			password: "",
+			isChecked: true,
+			formErrors: { username: "", password: "" },
+			usernameValid: false,
+			passwordValid: false,
+			formValid: false
+		};
+	}
+	handleUserInput = e => {
+		const name = e.target.name;
+		const value = e.target.value;
+		this.setState({ [name]: value }, () => {
+			this.validateField(name, value);
+		});
+		this.setState({ value: e.target.username });
+	};
+
+	handleChecked(e) {
+		this.setState({ isChecked: !this.state.isChecked });
+		console.log(this.state.isChecked);
+	}
+	validateField(fieldName, value) {
+		let fieldValidationErrors = this.state.formErrors;
+		let usernameValid = this.state.usernameValid;
+		let passwordValid = this.state.passwordValid;
+
+		switch (fieldName) {
+			case "username":
```

Comment:
Look into yup for form validation and tell me whether you'd like to use it.

Reply:
Sry, I didn’t understand what you mean.